src/components/auth/SignUp.tsx

```diff
@@ -0,0 +1,145 @@
+import { Form, Formik } from 'formik';
+import { PersonAdd as PersonAddIcon } from '@material-ui/icons';
+
+import { UserCreationData } from '@app/types';
+
+import { KU_EMAIL_REGEX } from '@app/constants';
+import { useUser } from '@app/contexts/services/User';
+
+import { FormikPasswordField, FormikSelectField, FormikTextField } from '../common/FormFields';
+
+import AuthContentContainer from './AuthContentContainer';
+
+import useStyles from './_jss/SignUp.jss';
+
+interface SignUpProps {
+  /**
+   * Callback triggered when a closng action is taken
+   */
+  onClose: () => void;
+  /**
+   * Callback triggered when the link to sign in is clicked
+   */
+  onSignInRedirect: () => void;
+}
+
+const SignUp = React.forwardRef<unknown, SignUpProps>((props, ref) => {
+  const { onClose, onSignInRedirect } = props;
+  const [error, setError] = useState<string | undefined>(undefined);
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const classes = useStyles();
+  const { signUp } = useUser();
+  const halls = [
+    'Battenfeld',
+    'Douthart',
+    'Grace Pearson',
+    'KK Amini',
+    'Krehbiel',
+    'Margaret Amini',
+    'Miller',
+    'Pearson',
+    'Rieger',
+    'Sellards',
+    'Stephenson',
+    'Watkins',
+  ];
+
+  const handleSubmit = useCallback(
+    async (data: UserCreationData) => {
+      setLoading(true);
+      try {
+        await signUp(data);
+        onClose();
+      } catch (e) {
+        setError(e.message);
+      }
+      setLoading(false);
+    },
+    [onClose, signUp],
+  );
+
+  return (
+    <AuthContentContainer
+      className={classes.signUpContainer}
+      error={error}
+      header={{
+        title: 'Sign Up',
+        icon: <PersonAddIcon />,
+        subtitle: 'Sign up using your KU email account',
+      }}
+      bottomText="Already have an account? Sign In"
+      isLoading={isLoading}
+      onClose={onClose}
+      onBottomTextClick={onSignInRedirect}
+      ref={ref}
+    >
+      <Formik
+        initialValues={
+          {
+            email: '',
+            password: '',
+            displayName: '',
+            hall: '',
+            roomNumber: undefined,
+          } as UserCreationData
+        }
+        validationSchema={Yup.object({
+          email: Yup.string()
+            .matches(KU_EMAIL_REGEX, 'Invalid KU email')
+            .required('Required'),
+          password: Yup.string()
+            .min(8, 'Password must be at least 8 characters')
+            .required('Required'),
+          displayName: Yup.string().required('Required'),
+          hall: Yup.string().required('Required'),
+          roomNumber: Yup.number()
+            .positive('Room number must be positive')
+            .integer('Room number must be a whole number'),
+        })}
+        onSubmit={handleSubmit}
+      >
+        <Form>
+          <FormikTextField name="email" label="KU Email" disabled={isLoading} required />
+          <FormikPasswordField
```

Comment:
I didn't add it because we have a visibility toggle so the user can check to see what they're typing. I personally don't much care for confirm password fields when filling out sign up forms, the fewer fields to fill out the better imo. https://www.viget.com/articles/password-fields-are-annoying/

Reply:
Ah, I sort of feel the exact opposite tbh. Most of my passwords are > 12 characters and I'm more re-assured by the low probability that I type them wrong twice than I am by being forced to have to look over every character with my own eyes. In most cases I think typing it twice is faster.

But honestly I don't have a strong preference either way, this works for me